test/remote/gateways/remote_safe_charge_test.rb

```diff
@@ -65,6 +65,15 @@ def test_successful_purchase
     assert_equal 'Success', response.message
   end
 
+  def test_successful_purchase_with_non_fractional_currency
+    options = @options.merge(currency: 'CLP')
+    response = @gateway.purchase(127999, @credit_card, options)
+
+    assert_success response
+    assert_equal 'Success', response.message
+    assert_equal '1279', response.params['requestedamount']
```

Comment:
My only thought on this is why did it round down when it is 1 cent away from being `1280`, but that has nothing to do with the code that you wrote 😆

Reply:
Yes, good question. I provide a description of this in the ticket for the task.

Basically, the `localized_amount` method also looks at the gateway adapter's `money_format`, and depending on whether that is set to `:cents` or not determines how it handles the extra digits.